src/main/java/org/mule/extension/http/internal/listener/HttpRequestAttributesResolver.java

```diff
@@ -74,11 +81,22 @@ public HttpRequestAttributes resolve() {
         .uriParams(decodeUriParams(listenerPath, rawPath))
         .queryString(queryString)
         .queryParams(decodeQueryString(rawQuery))
-        .localAddress(requestContext.getServerConnection().getLocalHostAddress().getAddress().getHostAddress().concat(":")
-            .concat(Integer.toString(requestContext.getServerConnection().getLocalHostAddress().getPort())))
-        .remoteAddress(clientConnection.getRemoteHostAddress().getHostString().concat(":")
-            .concat(Integer.toString(clientConnection.getRemoteHostAddress().getPort())))
+        .localAddress(resolveLocalAddress())
+        .remoteAddress(resolveRemoteAddress(clientConnection))
         .clientCertificate(clientConnection::getClientCertificate)
         .build();
   }
+
+  private String resolveLocalAddress() {
+    return remoteAddressFormat.equals(RemoteAddressFormat.HOSTNAME_AND_ADDRESS)
+        ? requestContext.getServerConnection().getLocalHostAddress().toString()
```

Comment:
This reminds me of the org.mule.extension.http.api.error.HttpError#UNSUPPORTED_MEDIA_TYPE enum. Let me know what you think.

Reply:
I am not following you